com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext

```diff
@@ -142,19 +134,19 @@ Operation:
 // @Compat deprecated - to be deleted once all models updated
 AssignOutputOperation:
 	'assign-output' assignRoot = [AssignPathRoot|ValidID] (path = Segment)? ':' =>Definable?
-	expression = RosettaCalcExpression assignAsKey?='as-key'?
+	expression = RosettaCalcExpressionWithAsKey
 ;
 
 OutputOperation:
 	('set' | add ?= 'add') assignRoot = [AssignPathRoot|ValidID] (path = Segment)? ':' =>Definable?
-		expression = RosettaCalcExpression assignAsKey?='as-key'?
+		expression = RosettaCalcExpressionWithAsKey
 ;
 
 Segment:
-	FEATURE_CALL_SEPARATOR attribute = [Attribute|ValidID] (next = Segment)?;
+	'->' attribute = [Attribute|ValidID] (next = Segment)?;
```

Comment:
Cleanup 3: I made `->` a keyword instead of a terminal rule. Makes it much easier to format.